lighthouse-core/test/config/config-test.js

```diff
@@ -265,6 +265,64 @@ describe('Config', () => {
     }), /meta.requiredArtifacts property/);
   });
 
+  it('filters the config', () => {
+    const config = new Config({
+      settings: {
+        onlyCategories: ['needed-category'],
+        onlyAudits: ['color-contrast'],
+      },
+      passes: [
+        {recordTrace: true, gatherers: []},
+        {recordNetwork: true, gatherers: ['accessibility']},
+      ],
+      audits: [
+        'accessibility/color-contrast',
+        'first-meaningful-paint',
+        'time-to-interactive',
+        'estimated-input-latency',
+      ],
+      categories: {
+        'needed-category': {
+          audits: [
+            {id: 'first-meaningful-paint'},
+            {id: 'time-to-interactive'},
+          ],
+        },
+        'other-category': {
+          audits: [
+            {id: 'color-contrast'},
+            {id: 'estimated-input-latency'},
+          ],
+        },
+        'unused-category': {
+          audits: [
+            {id: 'estimated-input-latency'},
+          ]
+        }
+      },
+    });
+
+    assert.ok(config.audits.length, 3);
+    assert.equal(config.passes.length, 2);
+    assert.ok(!config.categories['unused-category'], 'removes unused categories');
+    assert.equal(config.categories['needed-category'].audits.length, 2);
+    assert.equal(config.categories['other-category'].audits.length, 1);
+  });
+
+  it('filtering works with extension', () => {
+    const config = new Config({
+      extends: true,
+      settings: {
+        onlyCategories: ['performance'],
+        onlyAudits: ['is-on-https'],
+      },
+    });
+
+    assert.ok(config.audits.length, 'inherited audits by extension');
+    assert.ok(config.audits.length < origConfig.audits.length, 'filtered out audits');
```

Comment:
could you do `origConfig.categories.performance.audits.length + 1`? Seems future compatible